integration-tests/main/src/main/java/io/quarkiverse/googlecloudservices/it/StorageResource.java

```diff
@@ -6,26 +6,25 @@
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 
-import com.google.cloud.storage.Blob;
-import com.google.cloud.storage.BlobInfo;
-import com.google.cloud.storage.Bucket;
-import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.*;
 
 @Path("/storage")
 public class StorageResource {
     private static final String BUCKET = "quarkus-hello";
+    private static final String FILE = "test-upload.txt";
 
     @Inject
     Storage storage;
 
     @GET
-    @Produces(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.APPLICATION_JSON)
     public String storage() {
-        BlobInfo blobInfo = BlobInfo.newBuilder(BUCKET, "test-upload").build();
-        storage.create(blobInfo, "test".getBytes());
+        storage.create(BucketInfo.newBuilder(BUCKET).build());
+        BlobInfo blobInfo = BlobInfo.newBuilder(BUCKET, FILE).build();
+        storage.create(blobInfo, "{\"success\": true}".getBytes());
 
         Bucket bucket = storage.get(BUCKET);
-        Blob blob = bucket.get("hello.txt");
+        Blob blob = bucket.get(FILE);
         return new String(blob.getContent());
     }
```

Comment:
Please, keep the test idempotent, I need to be able to call this endpoint multiple time as it is used also during modification of the library on a real projet on the real storage runtime.

I'd rather : keep it as before and upload the file to the bucket from your test, or make it idempotent by deleting the buket at the end of the method.
You can also create a separate method for your test if you want but this will just adds complexity

Reply:
So is it intentional that the retrieved file is a different one from the uploaded one? I’ll see what’s best, since for the upload a fully instantiated storage instance is required. It’s possible to provide a data folder to the mock. 
So the current assumption is that a hello.txt file is already in the bucket when this method is called, correct?

Reply:
Yes, currently I use an existing bucket with an existing file inside my test.
My point here is that I should be able to launch the test or launch the apps and call the endpoint and both should works.
The test should use an embedded storage.
The apps should use a real storage instance from my google cloud project.